### DISCUSSION TITLE
Equation reference style (@eq -> eq. (1))

### DISCUSSION THREAD

**Discussion by Qottmann** [typst/typst#1917] [Q&A] [answered]
I would like to have references to equations in the style I am most familiar with, i.e. referencing an equation `@eq` yields "eq. (1)", as opposed to "Equation 1" which is the current default. Is this possible?

Thank you in advance!

**Comment by Julian-Wassmann** [comment, marked as answer]
Hi,
This is possible using a show rule for `ref`. Checking the element type is necessary so only equations (and not figures etc.) get the custom reference.
```
#set math.equation(
  numbering: "1",
  supplement: none,
)
#show ref: it => {
  // provide custom reference for equations
  if it.element != none and it.element.func() == math.equation {
    // optional: wrap inside link, so whole label is linked
    link(it.target)[eq.~(#it)]
  } else {
    it
  }
}

Reference to @eq.
$ a^2 + b^2 = c^2 $ <eq>
```

**Reply by Qottmann** [reply]
Amazing, thank you so much! (Small change to `numbering: "(1)"` to have equation numbers wrapped in brackets for in the references and in the enumerations in the paper 🙂 )

**Comment by jpssrocha** [comment]
I think that:

#set math.equation(
  numbering: "(1)",
  supplement: "eq.",
)

would work ... at least here in 2025 hehe

the documentation on `ref` was inducing me to use the approach above ... but then i saw this example with `supplement: none` and tried to use it. and it worked for my case.

**Comment by Ptitet** [comment]
In this show rule:

```typst
#show ref: it => {
  // provide custom reference for equations
  if it.element != none and it.element.func() == math.equation {
    // optional: wrap inside link, so whole label is linked
    link(it.target)[eq.~(#it)]
  } else {
    it
  }
}
```

The `[eq.~(#it)]` renders as `eq.~(Equation 1)`.

How can I get just the "numbering part", the `1` in `Equation 1`? I would like the ref to render as `(1)`.

Thanks!

**Reply by Julian-Wassmann** [reply]
Hi,
that's why you need the `supplement: none` in the equation set rule. This will remove the "Equation" before the numbering.
My original answer above still works.

**Reply by lcalllegari1** [reply]
you can alternatively do it like this, if you want to keep your function almost untouched:

```typst
#show ref: it => {
  // provide custom reference for equations
  if it.element != none and it.element.func() == math.equation {
    // optional: wrap inside link, so whole label is linked
    link(it.target)[eq.~#it.element.numbering]
  } else {
    it
  }
}
```

<img width="973" height="63" alt="image" src="https://github.com/user-attachments/assets/bf657bb0-e070-43c9-9d34-480081672773" />


That is, use the properties of `it` to only view the numbering part of the element.

I don't know if this is useful, maybe supplement is just more straightforward, but I tried this and it does not seem so bad.

**Reply by Ptitet** [reply]
@lcalllegari1 Thanks for your reply, but be careful to use `#it` only, because `#it.element.numbering` refer to the `numbering` field in the `#set math.equation(numbering: "(1)")`, so it will always render the same.

**Reply by lcalllegari1** [reply]
Ahh, I see, should have done a little more testing I guess. I will next time, sorry for any confusion and thanks for the remark!
